x-pack/plugins/ml/public/application/components/data_grid/common.ts

```diff
@@ -82,6 +83,37 @@ export const getFieldsFromKibanaIndexPattern = (indexPattern: IndexPattern): str
   return indexPatternFields;
 };
 
+/**
+ * Return a map of runtime_mappings for each of the index pattern field provided
+ * to provide in ES search queries
+ * @param indexPatternFields
+ * @param indexPattern
+ * @param clonedRuntimeMappings
+ */
+export const getRuntimeFieldsMapping = (
+  indexPatternFields: string[],
+  indexPattern: IndexPattern,
+  clonedRuntimeMappings?: { [key: string]: RuntimeField }
+) => {
+  if (!Array.isArray(indexPatternFields) || indexPattern === undefined) return {};
+  const ipRuntimeMappings = indexPattern.getComputedFields().runtimeFields;
+  let combinedRuntimeMappings: { [ipField: string]: any } = {};
```

Comment:
can we provide a better type here?

Reply:
in my WIP PR i have created a type:
```
export type RuntimeMappings = Record<string, RuntimeField>;
```
in https://github.com/elastic/kibana/blob/master/x-pack/plugins/ml/common/types/fields.ts

This could be added to this PR too.

Reply:
I ended up manually create the RuntimeMappings https://github.com/elastic/kibana/pull/90363/commits/2bd6b3c59df1f90780fdfe9229e31c94eb78323f here as the ones exported from `data` or `runtime_fields` plugins don't take string for `script` field.

Reply:
ok cool, that's odd that the `RuntimeField ` from the data plugin is incorrect.